tests/reliability/agent_connection/test_agent_connection.py

```diff
@@ -0,0 +1,93 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: reliability
+
+brief: Agents register their state in the file /var/ossec/var/run/wazuh-agentd.state. This state can vary between
+       `disconnected`, `connected` and `pending` states. Similarly, the manager records the number of `tcp_sessions`
+        in the file /var/ossec/var/run/wazuh-remoted.state. Under normal conditions, the agents should never disconnect
+        and the tcp_sessions should be equal to the number of connected agents on the node.
+
+tier: 0
+
+modules:
+    - agentd
+    - remoted
+
+components:
+    - agent
+    - manager
+
+daemons:
+    - wazuh-agentd
+    - wazuh-remoted
+
+os_platform:
+    - linux
+    - windows
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+    - macOS Server
+    - macOS Catalina
+    - macOS Sierra
+    - Windows XP
+    - Windows 7
+    - Windows 8
+    - Windows 10
+    - Windows Server 2003
+    - Windows Server 2012
+    - Windows Server 2016
+    - Windows Server 2019
+
+tags:
+    - remoted_agent_communication
+    - remoted
+    - agentd
+'''
+import pytest
+
+
+def test_agent_connection(get_report):
```

Comment:
We need to add the test documentation

Reply:
Done [6dc99b3](https://github.com/wazuh/wazuh-qa/pull/2612/commits/6dc99b3201c9bcca8c630ee8466bf134031fddcc)